src/Modules/UngradedModules.jl

```diff
@@ -221,12 +328,34 @@ parent_type(::Type{FreeModElem{T}}) where {T} = FreeMod{T}
 elem_type(::FreeMod{T}) where {T} = FreeModElem{T}
 parent_type(::FreeModElem{T}) where {T} = FreeMod{T}
 
-function show(io::IO, e::FreeModElem)
-  if length(e.coords) == 0
-    print(io, 0)
-    return
+function expressify(e::FreeModElem; context = nothing)
+  sum = Expr(:call, :+)
+  for (pos, val) in e.coords
+     # assuming e.parent.S is an array of strings/symbols
+     push!(sum.args, Expr(:call, :*, expressify(val, context = context), e.parent.S[pos]))
   end
-  print(io, join(["($val)*$(e.parent.S[pos])" for (pos, val) in e.coords], " + "))
+  return sum
+end
+@enable_all_show_via_expressify FreeModElem
+
+@doc Markdown.doc"""
+    show_as_vector(io::IO, e::FreeModElem)
+
+Shows the free module element `e` as a vector instead of a linear combination 
+of the standard basis.
+"""
```

Comment:
`show_as_vector` seems like a strange function to have or want. Wouldn't it be more idomatic to implement a function like `Vector(e::FreeModElem)` returning the dense row, and then you can just type `Vector(e)` at the repl and get the printing?

Reply:
You are completely right, I changed it.